src/Cache/Cache.php

```diff
@@ -143,22 +111,74 @@ public function expired(string $key): bool
 	}
 
 	/**
-	 * Checks when the cache has been created;
-	 * returns the creation timestamp on success
-	 * and false if the item does not exist
+	 * Flushes the entire cache and returns
+	 * whether the operation was successful;
+	 * this needs to be defined by the driver
 	 */
-	public function created(string $key): int|false
+	abstract public function flush(): bool;
+
+	/**
+	 * Gets an item from the cache
+	 *
+	 * <code>
+	 *   // get an item from the cache driver
+	 *   $value = $cache->get('value');
+	 *
+	 *   // return a default value if the requested item isn't cached
+	 *   $value = $cache->get('value', 'default value');
+	 * </code>
+	 */
+	public function get(string $key, mixed $default = null): mixed
 	{
-		// get the Value object
+		// get the Value
 		$value = $this->retrieve($key);
 
-		// check for a valid Value object
+		// check for a valid cache value
 		if ($value instanceof Value === false) {
-			return false;
+			return $default;
 		}
 
-		// return the expires timestamp
-		return $value->created();
+		// remove the item if it is expired
+		if ($value->expires() > 0 && time() >= $value->expires()) {
+			$this->remove($key);
+			return $default;
+		}
+
+		// return the pure value
+		return $value->value();
+	}
+
+	/**
+	 * Returns a value by either getting it from the cache
+	 * or via the callback function which then it stored in
+	 * the cache for future retrieval
+	 * @since 3.8.0
+	 */
+	public function getOrSet(
+		string $key,
+		Closure $result,
+		int $minutes = 0
+	): mixed {
+		$value  = $this->get($key);
+		$result = $value ?? $result();
+
+		if ($value === null) {
+			$this->set($key, $result, $minutes);
```

Comment:
If a user explicitly stores `null` in the cache (for whatever reason), this code will treat the value like it wasn't stored. For consistency and reliability, I think we should differentiate between these cases.

What we could do is to move the expiry check from `get()` into a new protected method like `retrieveValid(string $key): Value|null`. The `get()` method can then just do:

```php
$value = $this->retrieveValid($key);

if ($value) {
	return $value->value();
}

return $default;
```

(Note: something like `$this->retrieveValid($key)?->value() ?? $default` will again *not* work for the same reason.)

The logic in `getOrSet()` can then be changed to:

```suggestion
		$value  = $this->retrieveValid($key);
		$result = $value?->value();

		if ($value === null) {
			$result = $result();
			$this->set($key, $result, $minutes);
```

Reply:
I feel like `retrieveValid` would over engineer it a bit - and I'd find it hard to understand the difference as user. We use the pattern of `getOrSet` in a few places already - in situations where the cached value cannot be `null`. I think it would be a lot easier to just add a comment to the docblock that this cannot be used on a `null` cache value - which would be an extreme edge case anyways.

Reply:
Fine for me. :)